week-1-ts-ecosystem/w01e03-object-oriented-structure/src/types/product.ts

```diff
@@ -0,0 +1,17 @@
+import {TypeCart} from "../cart/Cart";
+
+
+export interface ProductType<T extends TypeCart> {
```

Comment:
Ten interface jest moim zdaniem niepotrzebny.

To klasa `Product` powinna reprezentować dane oraz typ.
Pamiętaj że `interface` opisuje kształt danych a `class` ma kształt + reprezentację.

Ale ponieważ określa kształt - to nie trzeba pisać "mirrrora" tego co robi klasa.

Zauważ że to mega kosztowne utrzymywać i klasę i interface realizujących to samo zadanie
i odwzorowujących podobny kształt danych.